### ISSUE TITLE
[BUG] Upgrade from 1.4.4 engine to 1.5.3 causes an error on some volumes

### ISSUE THREAD

**Issue by artur9010** [longhorn/longhorn#7137]
## Describe the bug (🐛 if you encounter this issue)

After upgrading a volume engine from 1.4.4 to 1.5.3 I got this error

```
Expansion Error: the expected size 10737418240 of engine pvc-3bc9fa3b-003c-4bde-a338-d4fdee3e3198-e-8d223236 should not be smaller than the current size 21474836480
Note: You can cancel the expansion to avoid volume crash
```

I haven't changed volume size either in dashboard or pvc definition. I got that error on 7 of 28 volumes.

## To Reproduce

1. Upgrade volume engine from 1.4.4 to 1.5.3

**Comment by artur9010** [comment]
Also when I tried to cancel volume expansion I got an message saying that volume is not expanding

```
unable to cancel expansion for volume pvc-51e2f90b-4309-4c88-96b0-ca17b7268612: volume expansion is not started
```

**Comment by innobead** [comment]
This is a known issue. Even though it has been fixed in 1.5.2, there would be a rare chance to hit this if you have live volume and upgrade from a previous version before 1.5.2. 

Please check https://longhorn.io/docs/1.5.3/deploy/important-notes/#offline-upgrade-required-to-fully-prevent-unexpected-replica-expansion to see how to recover this. 

cc @ejweber @ChanYiLin

**Comment by artur9010** [comment]
It mostly happend on small volumes, like the 10GB one


```
Expansion Error: the expected size 10737418240 of engine pvc-cce1e1db-60e9-44b2-8223-a89406ce810e-e-8b626935 should not be smaller than the current size 32212254720
```

pvc spec
```yaml
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: longhorn-duo
  volumeMode: Filesystem
  volumeName: pvc-cce1e1db-60e9-44b2-8223-a89406ce810e
```

pv spec
```yaml
spec:
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 10Gi
  claimRef:
    apiVersion: v1
    kind: PersistentVolumeClaim
    name: db-data
    namespace: [removed]
    resourceVersion: '23520047'
    uid: cce1e1db-60e9-44b2-8223-a89406ce810e
  csi:
    driver: driver.longhorn.io
    fsType: ext4
    volumeAttributes:
      numberOfReplicas: '2'
      storage.kubernetes.io/csiProvisionerIdentity: 1696599830025-8081-driver.longhorn.io
    volumeHandle: pvc-cce1e1db-60e9-44b2-8223-a89406ce810e
  persistentVolumeReclaimPolicy: Delete
  storageClassName: longhorn-duo
  volumeMode: Filesystem
```

but somehow the volume is 30GB?
```
root@mongodb-0:/# df -h
Filesystem                                              Size  Used Avail Use% Mounted on
overlay                                                 233G   85G  136G  39% /
tmpfs                                                    64M     0   64M   0% /dev
/dev/longhorn/pvc-cce1e1db-60e9-44b2-8223-a89406ce810e   30G  495M   29G   2% /data/db
```
